### ISSUE TITLE
Write EDC15 EEPROM

### ISSUE THREAD

**Issue by kbs1** [gmenounos/kw1281test#54]
Hi,

your tool is the only one that is able to dump the EDC15 EEPROM on my skoda octavia year 2001. I am using CH340 "blue" cable for vag-com 409.1. No other tool was able to dump the eeprom for me.

I have now modified the eeprom and would like to ask, if in future, there are plans to implement "write edc15 eeprom" functionality. That would be amazing.

Thank you very much,
keep up the great work! :)

**Comment by gmenounos** [comment]
Glad to hear that it worked for you. I do plan to add the ability to write the EEPROM at some point. It's not difficult but I'm concerned that if anything goes wrong that it would "brick" the ECU, making the car unusable.

**Comment by kbs1** [comment]
Thank you for the positive news, I think that might actually be possible to brick it that way. I am still experimenting with other tools and cables to maybe be able to write the eeprom. If it fails, I am prepared to disassemble the ECU and desolder the eeprom and use chip programmer to re-flash it.

**Comment by kbs1** [comment]
Hello, I was checking further, it seems the loader already contains code to write one eeprom byte. I only need to write ~4 bytes to disable the immo, which is intermittent.

It seems I could use something like 
```
kwp2000.SendMessage((DiagnosticService)0xA6, new byte[] { 0xFF }, excludeAddresses: true);
```
...to write one eeprom byte at a time? But how do I specify the address or offset in the page?

**Comment by gmenounos** [comment]
I will work on it soon. Also, which 4 bytes need to be changed? The ECUs that I have (038906012BD and 038906012GN) only require that 2 bytes be changed (locations $1B0 and $1DE both should be changed to $60).

**Comment by gmenounos** [comment]
Ok, this new release supports writing using the new WriteEdc15Eeprom command: https://github.com/gmenounos/kw1281test/releases/tag/v0.91-beta

For example (EDC15 Immobilizer Off):

`.\kw1281test.exe com2 9600 1 WriteEdc15Eeprom 0x1B0 0x60 0x1DE 0x60`

Give it a try and let me know if it works for you. If anything goes wrong during the writing process, you'll probably need to disconnect your battery for a minute to reset the ECU. A backup copy of the entire EEPROM will be created just before writing the new values.

**Comment by kbs1** [comment]
Thank you very much, it has worked perfectly! 👍 Now my car is immo-free and won't leave me stranded because of intermittent key reading. I also successfully deactivated our other family car, so it worked flawlessly on ECUs `038906012H ` and `038906019CD`.

This tool is the only one I found that can manipulate EDC15 EEPROM through standard OBD II connector without any further modifications to the car or any disassembly.

Bytes - you are right, the two bytes disabled the immo completely, I remembered it incorrectly. Thank you again! :)
